src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -79,6 +108,8 @@ public RequestExecutionQueue(ILspSolutionProvider solutionProvider)
             // Create a task completion source that will represent the processing of this request to the caller
             var completion = new TaskCompletionSource<TResponseType>();
 
+            // Note: If the queue is not accepting any more items then TryEnqueue below will fail.
+
             var textDocument = handler.GetTextDocumentIdentifier(request);
             var item = new QueueItem(mutatesSolutionState, clientCapabilities, clientName, textDocument,
                 callbackAsync: async (context, cancellationToken) =>
```

Comment:
Should the call to completion.SetCancelled in here be passing along the cancellation token so we can better diagnose if things are going wrong? You might have to Debug.Assert(completion.TrySetCancelled(...)) since I don't know if the overload was added.

Reply:
I didn't even know that overload existed! Looks like nothing else calls it in Debug.Assert so I'm just gonna go for it. Given that the completion tokens entire lifetime is in this method I don't know why it would ever fail, but I'm happy to educated.